### DISCUSSION TITLE
error in installing Libmesh On HPC

### DISCUSSION THREAD

**Discussion by gmaurya20** [idaholab/moose#22289] [Q&A General] [answered]
I have successfully installed PETSc using the offline installation method, but I am getting errors in the LIBMESH installation. The error is attached in the .txt file. please help me to solve this.
[text.txt](https://github.com/idaholab/moose/files/9704261/text.txt)

**Comment by gmaurya20** [comment]
To install PETSc, I have used :
python3.7.9
GCC-8.1.0
Mpich-4.0.2
cmake-3.23.1
Make-4.3

**Reply by milljm** [reply]
While I haven't read all the way through the text.txt attachment yet, I see you ran `./update_and_rebuild_libmesh.sh`. When running this script, it produces a `libmesh_diagnostic.log` file in the same location. Can you post that log as well? Thanks!

**Reply by gmaurya20** [reply]
libmesh_diagnostic.log file
[libmesh_diagnostic.log](https://github.com/idaholab/moose/files/9707160/libmesh_diagnostic.log)

**Comment by milljm** [comment]
I see something strange in the fact that the build immediately tried to use something outside of your name space:

```bash
/usr/bin/grep: /home/jsoni/GCC-8.1.0/lib/../lib64/libatomic.la: Permission denied
/usr/bin/sed: can't read /home/jsoni/GCC-8.1.0/lib/../lib64/libatomic.la: Permission denied
libtool:   error: '/home/jsoni/GCC-8.1.0/lib/../lib64/libatomic.la' is not a valid libtool archive
make[2]: *** [Makefile:744: libdbg.la] Error 1
```

There is a reference to `jsoni`'s home. Did this person perhaps give you their finished build of GCC? If so, I'm afraid it doesn't work this way for compilers. But I am also curious as to how PETSc was able to build and libMesh.

Edit: how PETSc was able to build and _not_ libMesh.

**Reply by gmaurya20** [reply]
configure and make log file for PETSc
[make.log.txt](https://github.com/idaholab/moose/files/9707261/make.log.txt)
[configure.log.txt](https://github.com/idaholab/moose/files/9707262/configure.log.txt)

**Reply by GiudGiud** [reply]
@gmaurya20 did you install the whole stack of compilers yourself?

**Reply by gmaurya20** [reply]
yes. I installed.

**Reply by milljm** [reply]
I don't think anyone here is going to know how to prevent the compiler from accessing this location:

```pre
/home/jsoni
```

This is a hard coded home directory path that _no one but jsoni has access to_. Your environment variables are clean. I know of no way to change GCC's behavior to not do this. I think you will need to rebuild GCC over again (and everything else you built with this compiler).

**Reply by gmaurya20** [reply]
Ok, 
I am installing GCC and MPICH again.

**Comment by roystgnr** [comment, marked as answer]
libatomic is not something we ever ask for explicitly, it's an internal library of GCC ... though it looks like it should be linked implicitly any time C11 or C++11 or higher standards are requested, which should be the default for gcc and g++ 8.1, so if you're using a hand-built gcc binary without having access to all its install directories then you could see an error like this.  It's also possible to have it linked implicitly because some dependency libtool file asked for it ... contrib/gzstream gives `-lz` to the linker, so if there's a libz.la somewhere around that is pointing to /some/wrong/directory/libatomic.la then that could also result in an error like this.

It's interesting that the failure only happens when trying to link a libtool library for gzstream, *after* there's already been a successful link of one for gmv.  Could be that the C compiler is working but the C++ compiler isn't - that would explain why gmv (which is pure C) and PETSc (which has C++ options but IIRC the C++ is usually turned off) could link when gzstream (which is pure C++) couldn't.

Speculations aside, no software hardcodes `/home/jsoni` anywhere.  There's got to be a hand-built gcc or a hand-built libz or *something* of that sort pointing to it.

**Reply by gmaurya20** [reply]
Thank you all,
Installing GCC and MPICH again---- It worked.
